test/e2e/config/caren.yaml

```diff
@@ -161,6 +184,31 @@ variables:
   AMI_LOOKUP_FORMAT: "konvoy-ami-{{.BaseOS}}-release-?{{.K8sVersion}}-*"
   AMI_LOOKUP_BASEOS: "rocky-9.1"
   AMI_LOOKUP_ORG: "999867407951"
+  # IP/FQDN of Prism Central.
+  NUTANIX_ENDPOINT: ""
```

Comment:
Thanks for the comments! What happens if a user doesn't set one of these empty envs, will it fail or does it accept an empty string as a valid input?

Reply:
we dont do anything special with these variables but pass along. its the capx which will report the errors and test may(should) fail

Reply:
we can explore validating webhook or CIL patterns in capx if not already

Reply:
So I just tested this, I think we should remove all the empty vars or maybe comment them out in this file.
With them set to an `""` the test proceeds (but fails actually creating the cluster).
But without the vars in this file, the e2e failure happens much earlier with a clear error before even attempting to launch a cluster.

```
  [FAILED] failed to run clusterctl init
  Unexpected error:
      <*errors.withStack | 0x14001292a38>: 
      failed to get provider components for the "nutanix:v1.4.0-alpha.2" provider: failed to perform variable substitution: value for variables [NUTANIX_ENDPOINT] is not set. Please set the value using os environment variables or the clusterctl config file
      {
          error: <*errors.withMessage | 0x14000610ae0>{
              cause: <*errors.withStack | 0x14001292a08>{
                  error: <*errors.withMessage | 0x14000610ac0>{
                      cause: <*yamlprocessor.errMissingVariables | 0x140012929d8>{
                          Missing: ["NUTANIX_ENDPOINT"],
                      },
                      msg: "failed to perform variable substitution",
                  },
```

Reply:
Similar message with a Cluster var
```
  [FAILED] Failed to run clusterctl config cluster
  Unexpected error:
      <*yamlprocessor.errMissingVariables | 0x14003482e70>: 
      value for variables [NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME] is not set. Please set the value using os environment variables or the clusterctl config file
      {
          Missing: [
              "NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME",
          ],
      }
```

Reply:
now commented.

Reply:
if we comment then github action e2e run always will fail 
<pre>
  Unexpected error:
      <*errors.withStack | 0xc0010a6198>: 
      failed to get provider components for the "nutanix:v1.4.0-alpha.2" provider: failed to perform variable substitution: value for variables [NUTANIX_ENDPOINT, NUTANIX_PASSWORD, NUTANIX_USER] is not set. Please set the value using os environment variables or the clusterctl config file
      {
          error: <*errors.withMessage | 0xc000ad9b20>{
              cause: <*errors.withStack | 0xc0010a6168>{
                  error: <*errors.withMessage | 0xc000ad9b00>{
                      cause: <*yamlprocessor.errMissingVariables | 0xc000013f68>{
                          Missing: ["NUTANIX_ENDPOINT", "NUTANIX_PASSWORD", "NUTANIX_USER"],
                      },
                      msg: "failed to perform variable substitution",
                  },
                  stack: [0x19d95c5, 0x19dc498, 0x1d94416, 0x1d99b5a, 0x1d99525, 0x1d98905, 0x1ee9433, 0x1eeb888, 0x1f6df77, 0x1f66207, 0x4b11a6, 0x4b02b9, 0xe5fe3e, 0xe7020e, 0xe73aad, 0x47b621],
              },
              msg: "failed to get provider components for the \"nutanix:v1.4.0-alpha.2\" provider",
          },
          stack: [0x1d99f6e, 0x1d99525, 0x1d98905, 0x1ee9433, 0x1eeb888, 0x1f6df77, 0x1f66207, 0x4b11a6, 0x4b02b9, 0xe5fe3e, 0xe7020e, 0xe73aad, 0x47b621],
      }
</pre>

Reply:
given its a initial checkin and once we have a setup, we will setup for e2e runs in github, we will set correct variables anyway. i dont think we should comment at this point. let me know wdyt?

Reply:
As discussed with @dkoshkin we have added clusterctl init required variables for nutanix as a secret in GH and workflow